src/lex/mod.rs

```diff
@@ -568,10 +550,8 @@ impl Lexer {
             }
         }
         let (term_err, newline_err) = (
-            Err(String::from(
-                "Missing terminating ' character in char literal",
-            )),
-            Err(String::from("Illegal newline while parsing char literal")),
+            Err(LexError::CharMissingEndQuote),
+            Err(LexError::NewlineInChar),
```

Comment:
I think you can get rid of this assignment in favor of just using the variants directly. I only had this here because the error message was long and I didn't want to type it more than once lol